tests/rest/devices.spec.js

```diff
@@ -0,0 +1,211 @@
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with an array instead of an object should return 400', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('[{"psk":"cHNrMQ==","psk_id":"cHNraWQx","uuid":"ABC"}, {"psk":"cHNrMg==","psk_id":"cHNraWQy","uuid":"DEF"}]')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with missing key in payload should return 400', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('{"psk_id":"cHNraWQx","uuid":"ABC"}')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with invalid base64 string should return 400', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('{"psk":"invalid-base64-string","psk_id":"cHNraWQx","uuid":"ABC"}')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with invalid value at key should return 400', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('{"psk":true,"psk_id":"cHNraWQx","uuid":"ABC"}')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with additional invalid key should return 201', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('{"psk":"cHNrNA==","psk_id":"cHNraWQ0","uuid":"JKL", "invalid-key":"invalid-value"}')
+        .end( (err, res) => {
+          should.not.exist(err);
+          res.should.have.status(201);
+          done();
+        });
+    });
+  });
+
+  describe('GET /devices', function() {
+    it('should return json array with four elements', (done) => {
+      chai.request(server)
+        .get('/devices')
+        .end((err, res) => {
+          should.not.exist(err);
+          res.should.have.status(200);
+          res.should.have.header('content-type', 'application/json');
+
+          res.body.should.be.a('array');
+          res.body.length.should.be.eql(4);
+
+          res.body[2].should.be.a('object');
+          res.body[2].should.have.property('psk_id');
+          res.body[2].psk_id.should.be.eql('cHNraWQy');
+
+          done();
+        });
+    });
+  });
+
+  describe('GET /devices:name', function() {
+    it('should return a single object', (done) => {
+      chai.request(server)
+        .get('/devices/ABC')
```

Comment:
It would make sense to save UUID from one of the test-generated device entries (in POST testing) and use it here, together with test-case device name. This way test will not depend on external file (database) contents.